web/src/features/time/TimeAxis.tsx

```diff
@@ -41,13 +47,34 @@ const renderTick = (
       opacity={1}
       transform={`translate(${scale(value)},0)`}
     >
+      {isMidnightTime && !isTimeController && (
+        <line
+          stroke="currentColor"
+          strokeDasharray="2,2"
+          y1={chartHeight ? -chartHeight : '-100%'}
+          y2="0"
+          opacity={0.6}
+          className="midnight-marker"
+        />
+      )}
       <line stroke="currentColor" y2="6" opacity={shouldShowValue ? 0.5 : 0.2} />
       {shouldShowValue &&
         renderTickValue(value, index, displayLive, lang, selectedTimeRange, timezone)}
     </g>
   );
 };
 
+const isMidnight = (date: Date, timezone?: string) => {
+  if (!timezone) {
+    return date.getHours() === 0 && date.getMinutes() === 0;
+  }
+
+  return (
+    new Date(date.toLocaleString('en-US', { timeZone: timezone })).getHours() === 0 &&
+    new Date(date.toLocaleString('en-US', { timeZone: timezone })).getMinutes() === 0
+  );
+};
```

Comment:
Question, why are these hardcoded to `en-US`?

Reply:
Good catch! I'll update it!